gherkin-utils/javascript/src/pretty.ts

```diff
@@ -1,78 +1,227 @@
 import * as messages from '@cucumber/messages'
 
-export default function pretty(gherkinDocument: messages.GherkinDocument): string {
+export type Syntax = 'markdown' | 'gherkin'
+
+export default function pretty(
```

Comment:
The pretty formatter for TS/JS was [recently given an official re-write](https://github.com/jbpros/cucumber-pretty-formatter) - could we not re-use that code here?

Reply:
Or am I reading it wrong and this is for pretty-printing markdown?

Reply:
This library converts a [GherkinDocument](https://github.com/cucumber/common/blob/main/messages/messages.md#gherkindocument) to a Gherkin or Markdown string, depending on what `Syntax` is passed as the 2nd argument.

The other pretty formatter has a different processing model - it prints lines as results are coming in. This library just prints the whole thing in one go.

I'd rather evolve this codebase to be able to print with results as well, but I'd rather handle that in a separate pull request.